lua/rip-substitute/utils.lua

```diff
@@ -1,6 +1,26 @@
 local M = {}
 --------------------------------------------------------------------------------
 
+---@generic T
+---@param list Array<`T`>
+---@param cb function(value: `T`,i:integer): `T`
+M.map = function(list, cb)
+	local result = {}
+	for i, value in ipairs(list) do
+		table.insert(result, cb(value, i))
+	end
+	return result
+end
+
+---@param table table
+---@param cb function(value: any): boolean
+function M.index_of(table, cb)
```

Comment:
If this case helper function is only used in the `matches` module, I think helper functions like these can simply be moved to that module?

(btw, not camelCase)

Reply:
yeah, I guess.
I didn't know vim already had a lot of these higher order table functions etc. so I used to always have them in a utils file when developing a plugin.

I put it into matches and also made it less generic and therefore more concise.